market_analysis/views/default.py

```diff
@@ -0,0 +1,450 @@
+from pyramid.response import Response
+from pyramid.view import view_config
+from pyramid.httpexceptions import HTTPFound
+from pyramid.security import remember, forget
+from sqlalchemy.exc import DBAPIError
+from sqlalchemy.orm.exc import UnmappedInstanceError
+from sqlalchemy import and_, or_
+try:
+    from urllib.parse import urlencode
+except ImportError:     # pragma: no cover
+    from urllib import urlencode
+from ..models import Stocks, Users, Association
+from ..security import check_credentials
+from passlib.apps import custom_app_context as pwd_context
+import datetime
+import requests
+
+
+@view_config(route_name='search', renderer='../templates/search.jinja2',
+                        permission='secret')
+def search_stocks(request):
+    msg = 'Search for stocks that you would like to watch.'
+    search_results = []
+    if request.method == 'GET':
+        return {'stocks': search_results, 'msg': msg}
+    elif request.method == 'POST':
+        try:
+            search = request.params.get('search')
+            search1 = search.lower().capitalize()
+            search2 = search.upper()
+            search_query = request.dbsession.query(Stocks)\
+                .filter(or_(Stocks.name.startswith(search1),
+                        Stocks.symbol.startswith(search2)))
+        except DBAPIError:  # pragma: no cover
+            return Response(db_err_msg, content_type='text/plain', status=500)
+
+        for row in search_query:
+            search_results.append(row)
+        if len(search_results) == 0:
+            msg = 'No results found, try again.'
+        return {'stocks': search_results, 'msg': msg}
```

Comment:
Generally, we do not provide alternate conditional branches for GET and POST requests.  You have this pattern:

```python
do_common_stuff()
if request.method == 'GET'
    do_get_method_stuff()
    return get_data
elif request.method == 'POST'
    do_post_method_stuff()
    return post_response_data()
```

But a more experienced programmer will notice that that extra conditional is not required:

```python
do_common_stuff()
if request.method == 'POST':
    do_post_method_stuff()
    return post_response_data()
do_get_method_stuff()
return get_data
```

It's a nit to pick, but it's a mark of experience.